BuildConfigGen/Preprocessor.cs

```diff
@@ -17,7 +17,7 @@ internal partial class Preprocessor
         private static partial Regex elseAndEndIfPreprocess();
 
 
-        internal static void Preprocess(string file, IEnumerable<string> lines, ISet<string> configreprocessorVariableName, string configName, out string processedOutput, out List<string> validationErrors, out bool madeChanges)
+        internal static void Preprocess(string file, IEnumerable<string> lines, ISet<string> validConfigPreprocessorVariableNames, string configName, out string processedOutput, out List<string> validationErrors, out bool madeChanges)
```

Comment:
Logging a comment to Hold on this change, as we consider another option (server side re-direct instead of fallback in task/agent)